### PR TITLE
[ET-1568] ticket order

### PR DESCRIPTION
**Pull request by ScottGarrison**
### 🎫 Ticket

[[ET-1568](https://theeventscalendar.atlassian.net/browse/ET-1568)]

### 🗒️ Description

- In order to replicate this, both ET and ET+ seem to need to be activated.

Issue: When you first create Tickets and have them set in a specific order, once you edit one (like price for example), the order of the tickets gets swapped without manually saying to do so.

Solution: Making sure to get the existing order value of the ticket, and updating it when pushed to the database through params.

### 🎥 Artifacts 
https://www.loom.com/share/feee2caaf213447dadf51a750c02fda7

### ✔️ Checklist
- [ ✔️] I've included a changelog entry in the readme.txt file. 
- [✔️ ] My code is tested. 
- [✔️ ] My code has [proper inline documentation](https://the-events-calendar.github.io/products-engineering/docs/code-standards/).

### CODE REVIEW

File: src/resources/js/tickets.js [hunk 1]
```diff
@@ -637,16 +637,21 @@ var ticketHeaderImage = window.ticketHeaderImage || {};
 
 		$tribe_tickets.trigger( 'pre-save-ticket.tribe', e );
 
-		var $orders = $base_panel.find( '.tribe-ticket-field-order' );
+		var $ticketID = $edit_panel.find( '#ticket_id' ).val();
+		var $editParent = $base_panel.find( `[data-ticket-type-id="${$ticketID}"]` );
```
Comment (codingmusician) [inline src/resources/js/tickets.js:641]: Only use the `$` on jQuery objects. In this case, `ticketID` is a string.
```suggestion
		var ticketID = $edit_panel.find( '#ticket_id' ).val();
		var $editParent = $base_panel.find( `[data-ticket-type-id="${ticketID}"]` );
```

File: src/resources/js/tickets.js [hunk 2]
```diff
@@ -637,16 +637,21 @@ var ticketHeaderImage = window.ticketHeaderImage || {};
...
+		var $orders = $editParent.find( '.tribe-ticket-field-order' ).val();
```
Comment (codingmusician) [inline src/resources/js/tickets.js:642]:
```suggestion
		var orders = $editParent.find( '.tribe-ticket-field-order' ).val();
```

File: src/resources/js/tickets.js [hunk 3]
```diff
@@ -637,16 +637,21 @@ var ticketHeaderImage = window.ticketHeaderImage || {};
...
 			data: $edit_panel.find( 'input,textarea,select' ).serialize().replace( /\'/g, '%27' ).replace( /\:/g, '%3A' ),
 			post_id: $post_id.val(),
 			nonce: TribeTickets.add_ticket_nonce,
-			menu_order: $orders.length,
+			menu_order: $orders,
 			is_admin: $( 'body' ).hasClass( 'wp-admin' )
 		};
 
+		// ticket_menu_order is missing from the serialized string, lets add it
+		params.data = params.data.concat( "&ticket_menu_order=" + $orders )
```
Comment (codingmusician) [inline src/resources/js/tickets.js:653]:
```suggestion
			menu_order: orders,
			is_admin: $( 'body' ).hasClass( 'wp-admin' )
		};

		// ticket_menu_order is missing from the serialized string, lets add it
		params.data = params.data.concat( "&ticket_menu_order=" + orders )
```

File: src/resources/js/tickets.js (hunk 1 above)
Comment (juanfra) [inline src/resources/js/tickets.js:641]: We should be ready now to use `let` and/or `const`

**Review by juanfra — COMMENTED** [review]
Great stuff 🥇

File: src/resources/js/tickets.js (hunk 1 above)
Comment (ScottGarrison) [reply src/resources/js/tickets.js:641]: yes nice catch

Comment (ScottGarrison) [reply src/resources/js/tickets.js:641]: that makes sense but i figured it was better to keep the files themselves consistent. So use `let` and `const` in v2 files but existing v1 files, use whatever the file is currently using.

Let me know if you want me to update this anyways

**Review by bordoni — APPROVED** [review]

**Comment by juanfra** [comment]
Taking this to the finish line. @bordoni @codingmusician @rafsuntaskin  can I please get a CR?

**Review by rafsuntaskin — DISMISSED** [review]

**Review by rafsuntaskin — APPROVED** [review]

**Pull request merged**